torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -5200,6 +5200,19 @@ Tensor lu_factor_ex_jvp(
   }
 }
 
+Tensor logsumexp_jvp(const Tensor& self_p, const Tensor& self_t, IntArrayRef dim, bool keepdim) {
```

Comment:
I'm not sure if this is ok in terms of numerical stability.
I would expect this to have similar properties as the forward and I don't think the current implementation does.
Does the OpInfo have test with large values for the input?

Reply:
> Does the OpInfo have test with large values for the input?

Nope, but I can try to add.

> I'm not sure if this is ok in terms of numerical stability.

Yeah good catch, I'm trying to implement the exp normalize trick. It seems to work in python, but still investigating why the exact same code breaks in c++:
```
import torch

self_p = torch.rand(3, 4, 5)
self_t = torch.rand(3, 4, 5)

dim = (1, 2)
keepdim = False

def logsumexp_jvp_stable(self_p, self_t, dim, keepdim):
    self_p_exp = (self_p - torch.amax(self_p, dim, True)).exp()
    sumexp_p = self_p_exp.sum(dim, keepdim)

    return (self_p_exp * self_t).sum(dim, keepdim) / sumexp_p

def logsumexp_jvp(self_p, self_t, dim, keepdim):
    self_p_exp = self_p.exp()
    sumexp_p = self_p_exp.sum(dim, keepdim)

    return (self_p_exp * self_t).sum(dim, keepdim) / sumexp_p

print(torch.allclose(logsumexp_jvp(self_p, self_t, dim, keepdim), logsumexp_jvp_stable(self_p, self_t, dim, keepdim)))

class LogSumExp(torch.autograd.Function):
    @staticmethod
    def forward(ctx, x, dim, keepdim):
        ctx.dim = dim
        ctx.keepdim = keepdim
        ctx.save_for_forward(x)
        return torch.logsumexp(x, dim, keepdim)

    @staticmethod
    def jvp(ctx, x_t, _1, _2):
        dim = ctx.dim
        keepdim = ctx.keepdim
        x, = ctx.saved_tensors
        return logsumexp_jvp_stable(x, x_t, dim, keepdim)

a = torch.rand(2, 4, 5, 6, requires_grad=True, dtype=torch.double) * 1e6

print(torch.autograd.gradcheck(LogSumExp.apply, (a, (1, 2), True),
      check_forward_ad=True, check_backward_ad=False, check_batched_grad=False))
```